File: spring-data-requery-java/src/test/java/com/coupang/springframework/data/requery/java/domain/basic/AbstractBasicGroup.java
package com.coupang.springframework.data.requery.java.domain.basic;

import com.coupang.kotlinx.data.requery.conveters.ByteArrayBlobConverter;
import com.coupang.springframework.data.requery.domain.AbstractPersistable;
import io.requery.*;
import io.requery.query.MutableResult;
import lombok.Getter;

/**
 * com.coupang.springframework.data.requery.java.domain.basic.AbstractBasicGroup
 *
 * @author debop
 * @since 18. 6. 4
 */
@Getter
@Entity
@Table(name = "basic_group")
public class AbstractBasicGroup extends AbstractPersistable<Integer> {

    private static final long serialVersionUID = 6213359648179473134L;

    @Key
    @Generated
    protected Integer id;

    @Column(unique = true)
    protected String name;

    protected String description;

    @Convert(ByteArrayBlobConverter.class)
    protected byte[] picture;

    @JunctionTable
    @ManyToMany
    protected MutableResult<AbstractBasicUser> members;
}

File: spring-data-requery-kotlin/src/test/kotlin/com/coupang/springframework/data/requery/domain/basic/AbstractBasicGroup.kt
package com.coupang.springframework.data.requery.domain.basic

import com.coupang.kotlinx.core.hashOf
import com.coupang.kotlinx.data.requery.conveters.ByteArrayBlobConverter
import com.coupang.kotlinx.objectx.ToStringBuilder
import com.coupang.springframework.data.requery.domain.AbstractPersistable
import io.requery.*
import io.requery.query.MutableResult

@Entity(cacheable = false, copyable = true)
abstract class AbstractBasicGroup: AbstractPersistable<Long>() {

    @get:Key
    @get:Generated
    @get:Column(name = "groupId")
    abstract val id: Long?

    @get:Column(nullable = false, unique = true)
    abstract var name: String

    abstract var description: String?

    @get:Convert(ByteArrayBlobConverter::class)
    abstract var picture: ByteArray?

    @get:JunctionTable
    @get:ManyToMany
    abstract val members: MutableResult<AbstractBasicUser>

    override fun hashCode(): Int {
        return hashOf(name)
    }

    override fun buildStringHelper(): ToStringBuilder {
        return super.buildStringHelper()
            .add("name", name)
    }
}